packages/flutter/lib/src/widgets/splash.dart

```diff
@@ -0,0 +1,621 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'framework.dart';
+import 'lookup_boundary.dart';
+import 'notification_listener.dart';
+import 'ticker_provider.dart';
+
+// Examples can assume:
+// late Color color;
+// late BuildContext context;
+
+/// Signature for the callback used by [Splash]es to obtain the
+/// appropriate [Rect].
+typedef RectCallback = Rect Function();
+
+/// A splash of [color] shown in response to a user gesture.
```

Comment:
I was a bit annoyed when I see the word "splash" used to explain the widget "splash".

What about "An animating shape shown in response to a user gesture."?

Reply:
Sounds good! [resolving here to consolidate conversation to [1 thread](https://github.com/flutter/flutter/pull/149963/files#r1672903573)]